python/packages/zarathustra/skills/subgraph_query_abci/behaviours.py

```diff
@@ -128,23 +150,42 @@ def async_act(self) -> Generator:
 
     def _request_subgraph_data(self):
         """Perform a http request to the subgraph api."""
-        self.context.logger.info("Requesting subgraph data.")
+
+        def try_obtaining_data(response) -> Optional[dict]:
+            try:
+                return json.loads(response.body)["data"]
+            except Exception as e:
+                self.context.logger.error(f"Failed to obtain subgraph query response {response}: {e}")
+                return None
+
+        def parse_responses(responses) -> dict[str, list[dict | None]]:
+            return {
+                name: list(map(try_obtaining_data, query_responses))
+                for name, query_responses in responses.items()
+            }
+
+        self.context.logger.info("Requesting subgraphs data.")
         api_key = self.params.config["subgraph_api_key"]
-        subgraph_config = json.loads(self.synchronized_data.most_voted_subgraph_config)
-        subgraph_url = subgraph_config["subgraph_url"]
-        query = subgraph_config["subgraph_query"]
+        subgraph_config = self.synchronized_data.most_voted_subgraph_config
+        subgraph_queries = subgraph_config["subgraph_queries"]
         headers = {"Content-Type": "application/json"}
-        url = subgraph_url.format(api_key=api_key)
-        content = json.dumps({"query": query}).encode("utf-8")
-        response = yield from self.get_http_response(
-            method="POST",
-            url=url,
-            content=content,
-            headers=headers,
-        )
-        data = json.loads(response.body)["data"]
-        self.context.logger.info(f"Subgraph query response: {data}")
-        return data
+
+        responses = {}
+        for name, config in subgraph_queries.items():
+            url = config.url.format(api_key=api_key)
+            for query in config.queries:
+                content = serialize({"query": query}).encode("utf-8")
+                response = yield from self.get_http_response(
+                    method="POST",
+                    url=url,
+                    content=content,
+                    headers=headers,
+                )
+                responses.setdefault(name, []).append(response)
```

Comment:
so ideally this would not be a nested loop, however that would require combining all queries into a single one, and I don't know how this can be achieved without merging the queries into one graphql query. But that requires manipulation of the individual queries, as described in my previous [comment here](https://github.com/Karrenbelt/verification-station/pull/38/files#r1720094324)